xcube/webapi/datasets/context.py

```diff
@@ -592,26 +592,42 @@ def _open_ml_dataset(self, dataset_config: DatasetConfig) -> MultiLevelDataset:
             data_store_pool = self.get_data_store_pool()
             data_store = data_store_pool.get_store(store_instance_id)
             data_id = dataset_config.get("Path")
-            open_params = dataset_config.get("StoreOpenParams") or {}
-            # Inject chunk_cache_capacity into open parameters
+            open_params = dict(dataset_config.get("StoreOpenParams") or {})
+            open_params_schema = data_store.get_open_data_params_schema(data_id=data_id)
+
+            # Inject cache_size=chunk_cache_capacity, if given and possible
             chunk_cache_capacity = self.get_dataset_chunk_cache_capacity(dataset_config)
             if (
                 chunk_cache_capacity
-                and (data_id.endswith(".zarr") or data_id.endswith(".levels"))
                 and "cache_size" not in open_params
+                and "cache_size" in open_params_schema.properties
             ):
                 open_params["cache_size"] = chunk_cache_capacity
+
+            # Inject data_type="mldataset", if possible
+            if (
+                "data_type" not in open_params
+                and "data_type" in open_params_schema.properties
+                and "mldataset" in data_store.get_data_types()
+            ):
+                open_params["data_type"] = "mldataset"
+
             with self.measure_time(
                 tag=f"Opened dataset {ds_id!r}"
                 f" from data store"
                 f" {store_instance_id!r}"
             ):
                 dataset = data_store.open_data(data_id, **open_params)
+
             if isinstance(dataset, MultiLevelDataset):
+                # Expected, nominal case.
                 ml_dataset: MultiLevelDataset = dataset
             else:
+                # Fallback. Usually results in poor tile computation performance.
                 ml_dataset = BaseMultiLevelDataset(dataset)
+
             ml_dataset.ds_id = ds_id
+
         else:
```

Comment:
As far as I understand this is the case for datasets calculated on the fly (in memory). Do we need to take care, that this is done as mldatasets as well, or is this covered already?